zcash_client_sqlite/src/wallet/common.rs

```diff
@@ -180,27 +183,210 @@ pub(crate) fn select_spendable_notes<P: consensus::Parameters, F, Note>(
+         AND rn.value > :min_value
+         AND accounts.ufvk IS NOT NULL
+         AND recipient_key_scope IS NOT NULL
+         AND nf IS NOT NULL
+         AND (
+            t.block IS NOT NULL -- the receiving tx is mined
+            OR t.expiry_height IS NULL -- the receiving tx will not expire
+            OR t.expiry_height >= :target_height -- the receiving tx is unexpired
+         )
+         AND rn.id NOT IN rarray(:exclude)
+         AND rn.id NOT IN (
+           SELECT {table_prefix}_received_note_id
+           FROM {table_prefix}_received_note_spends rns
+           JOIN transactions stx ON stx.id_tx = rns.transaction_id
+           WHERE stx.block IS NOT NULL -- the spending tx is mined
+           OR stx.expiry_height IS NULL -- the spending tx will not expire
+           OR stx.expiry_height >= :target_height -- the spending tx is unexpired
+         )
+         GROUP BY
+            rn.id, t.txid, rn.{output_index_col},
+            rn.diversifier, rn.value, {note_reconstruction_cols}, rn.commitment_tree_position,
+            ufvk, rn.recipient_key_scope, 
+            t.block, scan_state.max_priority"
+    ))?;
+
+    let excluded: Vec<Value> = exclude
+        .iter()
+        .filter_map(|ReceivedNoteId(p, n)| {
+            if *p == protocol {
+                Some(Value::from(*n))
+            } else {
+                None
+            }
+        })
+        .collect();
+    let excluded_ptr = Rc::new(excluded);
+
+    let row_results = stmt_select_notes.query_and_then(
+        named_params![
+            ":account_uuid": account.0,
+            ":target_height": &u32::from(target_height),
+            ":exclude": &excluded_ptr,
+            ":min_value": u64::from(zip317::MARGINAL_FEE)
+        ],
+        |row| -> Result<_, SqliteClientError> {
+            let result_note = to_spendable_note(params, row)?;
+            let max_priority_raw = row.get::<_, Option<i64>>("max_priority")?;
+            let shard_scan_priority = max_priority_raw
+                .map(|code| {
+                    parse_priority_code(code).ok_or_else(|| {
+                        SqliteClientError::CorruptedData(format!(
+                            "Priority code {code} not recognized."
+                        ))
+                    })
+                })
+                .transpose()?;
+
+            Ok((result_note, shard_scan_priority))
+        },
+    )?;
+
+    let trusted_height = target_height.saturating_sub(u32::from(confirmations_policy.trusted()));
+    let untrusted_height =
+        target_height.saturating_sub(u32::from(confirmations_policy.untrusted()));
+    // when attempting to select max spendable funds it's assumed that there are unconfirmed funds
+    // whereas when the caller is intending to spend `Everything` the presence of unmined or
+    // unconfirmed funds must cause a failure.
+    let attempts_max_spendable = matches!(max_spend_mode, MaxSpendMode::MaxSpendable);
```

Comment:
I find this documentation confusing. Nothing is "assumed" when the caller specified `MaxSpendable` - also, I think the way that this condition is being used in the `||` clause below is wrong.
```suggestion
    // when attempting to select max spendable funds it's assumed that there are unconfirmed funds
    // whereas when the caller is intending to spend `Everything` the presence of unmined or
    // unconfirmed funds must cause a failure.
    let attempts_max_spendable = matches!(max_spend_mode, MaxSpendMode::MaxSpendable);
```